libs/ngx-mime/src/lib/core/mime-resize-service/mime-resize.service.ts

```diff
@@ -21,25 +20,26 @@ export class MimeResizeService {
   }
 
   get onResize(): Observable<Dimensions> {
-    return this.resizeSubject.asObservable();
+    return this.resizeSubject.pipe(
+      debounceTime(200),
+      map((contentRect: DOMRectReadOnly) => {
+        return {
+          bottom: contentRect.bottom,
+          height: contentRect.height,
+          left: contentRect.left,
+          right: contentRect.right,
+          top: contentRect.top,
+          width: contentRect.width,
+        };
+      })
+    );
   }
 
-  markForCheck(): void {
-    if (!this.el) {
-      throw new Error('No element!');
-    }
-    const dimensions = this.mimeDomHelper.getBoundingClientRect(this.el);
-
-    if (
-      this.dimensions.bottom !== dimensions.bottom ||
-      this.dimensions.height !== dimensions.height ||
-      this.dimensions.left !== dimensions.left ||
-      this.dimensions.right !== dimensions.right ||
-      this.dimensions.top !== dimensions.top ||
-      this.dimensions.width !== dimensions.width
-    ) {
-      this.dimensions = dimensions;
-      this.resizeSubject.next({ ...this.dimensions });
-    }
+  initialize() {
+    const observer = new ResizeObserver((entry) => {
+      this.resizeSubject.next(entry[0].contentRect);
+    });
+    const el = <any>this.el.nativeElement.querySelector('#ngx-mime-mimeViewer');
```

Comment:
Do you need to cast to any here? EsLint will throw an error when we turn on the `@typescript-eslint/no-explicit-any` rule in the future.